library/common/http/client.cc

```diff
@@ -325,7 +326,9 @@ envoy_stream_intel Client::DirectStreamCallbacks::streamIntel() {
 
 void Client::DirectStream::saveLatestStreamIntel() {
   const auto& info = request_decoder_->streamInfo();
-  stream_intel_.connection_id = info.upstreamConnectionId().value_or(-1);
+  if (info.upstreamInfo()) {
```

Comment:
Do we need to set `connection_id` to `-1` in the case this fails to match previous behavior?

Reply:
yeah I added an initializer in the .h file to -1 :-)